### PR TITLE
Add the .listify() internal helper function

### LINKED ISSUE #299: add helper function to replicate argument as a list
We often have code like this when a function self-calls:

```R
if(is(dose_points, "list")){
  dose_points <- rep(dose_points, length = length(object))
}else{
  dose_points <- rep(list(dose_points), length = length(object))
}
```
We could have a simple helper function that allows to write the following:
```R
dose_points <- .listify(dose_points, length = length(object))
```
implemented along these lines:
```R
.listify <- function(variable, length) {
  if (!is.list(variable))
    variable <- list(variable)
  rep(variable, length = length)
}
```

### PR DESCRIPTION
**Pull request by mcol**
This adds the very simple `.listify()` helper, with essentially the implementation as posted in the issue, with the only difference of using `inherits()` instead of `is()`. Fixes #299.

### CODE REVIEW

File: R/plot_RLum.R [hunk 1]
```diff
@@ -126,25 +124,15 @@ plot_RLum<- function(
         mtext <- rep(list(...)$mtext, length.out = length(object))
```
Comment (RLumSK) [inline R/plot_RLum.R:124]: What about this one?

**Review by RLumSK — COMMENTED** [review]
I left one question and I hope that it works, I think not everywhere it makes sense but I see the consistency so we will try this

File: R/plot_RLum.R (hunk 1 above)
Comment (mcol) [reply R/plot_RLum.R:124]: There it's not creating a list. I tried to replicate the current behaviour, but I could convert that case too.

**Comment by mcol** [comment]
Where do you think that it doesn't make sense? It may be worth ironing those out now before we forget.

**Comment by RLumSK** [comment]
> Where do you think that it doesn't make sense? It may be worth ironing those out now before we forget.

I do remember that we had certain cases where this replication does not work so straight forward. I looked up the code of the other function and you see that we have some exceptions. Means, we have to ensure that what we expand, for instance, does not allow for `NULL` or something similar.  

 https://github.com/R-Lum/Luminescence/blob/acdaaa8218bf57b185e08418ca6e888db0111a9d/R/internals_RLum.R#L697-L727

File: R/plot_RLum.R (hunk 1 above)
Comment (mcol) [reply R/plot_RLum.R:124]: A clearer explanation: in the `main` case, we were sometimes creating a list (`main.list`) and depending on whether that `main.list` was created we were passing the corresponding element to `RLum.dispatcher()`. I decided to convert it always to a list so we could simplify the dispatch code (see the next hunk, lines 129-142). For `mtext` we were not creating a list in the first place, so I left it untouched.

**Comment by mcol** [comment]
Ah OK. Then this PR is not really affected, as I'm converting only the places where we were building lists manually.

**Pull request merged**